Helper/Search/Search.php

```diff
@@ -189,25 +201,55 @@ private function getOptions(ClientRequest $clientRequest): array
         throw new \LogicException('no search defined!');
     }
 
-    private function bindLimit(int $l): int
+    private function setFields(array $fields, string $locale): void
     {
-        if (null == $this->sizes) {
-            @trigger_error('Define allow sizes with the search option "sizes"', \E_USER_DEPRECATED);
-            return $l > 0 ? $l : $this->limit;
+        $this->fields = array_map(function (string $field) use ($locale) {
+            return str_replace('%locale%', $locale, $field);
+        }, $fields);
+    }
+
+    public function setSorts(array $data, string $locale): void
+    {
+        foreach ($data as $name => $options) {
+            if (\is_array($options)) {
+                $options['field'] = str_replace('%locale%', $locale, $options['field']);
+                $this->sorts[$name] = $options;
+            } elseif (\is_string($options)) {
+                $this->sorts[$name] = ['field' => str_replace('%locale%', $locale, $options)];
+            }
         }
+    }
 
-        if (\in_array($l, $this->sizes)) {
-            return $l;
-        } else {
-            return array_shift($this->sizes);
+    private function setSortBy(?string $name)
+    {
+        if (null === $name) {
+            return;
+        }
+
+        if (null == $this->sorts) {
+            @trigger_error('Define possible sort fields with the search option "sorts"', \E_USER_DEPRECATED);
+            $this->sortBy = $name;
+        } elseif (\array_key_exists($name, $this->sorts)) {
```

Comment:
can $this->sorts contain key value paris with value === null?
If so, use isset, except if you want to pass the null value?

Reply:
discussed